mlflow/pytorch/torchserve/deploy/serve.py

```diff
@@ -0,0 +1,496 @@
+        inference_port = "http://localhost:8080"
+        management_port = "http://localhost:8081"
+        address_strings = self.server_config["torchserve_address_names"]
+        if config_properties is not None and os.path.exists(config_properties):
+            with open(config_properties, "r") as f:
+                lines = f.readlines()
+                for line in lines:
+                    name = line.strip().split("=")
+                    if name[0] == address_strings[0] and name[1] is not None:
+                        inference_port = name[1]
+                    if name[0] == address_strings[1] and name[1] is not None:
+                        management_port = name[1]
+        return inference_port, management_port
+
+    def __validate_mandatory_arguments(self):
+        """
+        Validate the mandatory arguments is present if not raise exception
+        """
+
+        if "model_file" not in self.server_config:
+            raise Exception("Config Variable MODEL_FILE - missing")
+
+        if "handler_file" not in self.server_config:
+            raise Exception("Config Variable HANDLER_FILE - missing")
+
+    def create_deployment(self, name, model_uri, flavor=None, config=None):
+        """
+        Deploy the model at the model_uri to the specified target
+
+        :param name: Name of the of the model
+        :param model_uri: Serialized python file [.pt or .pth]
+        :param flavor: Flavor of the deployed model
+        :param config: Configuration parameters like model file path, handler path
+
+        :return: output - Returns a dict with flavor and name as keys
+        """
+
+        version = 1.0
+        is_version_provided = False
+        if config:
+            for key in config:
+                if key.upper() == "VERSION":
+                    version = float(config[key])
+                    is_version_provided = True
+                self.server_config[key.lower()] = str(config[key])
+
+        self.__validate_mandatory_arguments()
+
+        if not is_version_provided:
+            version = self.__get_max_version(name)
+
+        if "extra_files" not in self.server_config:
+            self.server_config["extra_files"] = None
+
+        if "requirements" not in self.server_config:
+            self.server_config["requirements"] = None
+
+        mar_file_path = self.__generate_mar_file(
+            model_name=name,
+            version=str(version),
+            model_file=self.server_config["model_file"],
+            handler_file=self.server_config["handler_file"],
+            requirements=self.server_config["requirements"],
+            extra_files=self.server_config["extra_files"],
+            model_uri=model_uri
+        )
+
+        config_registration = {
+            key: val
+            for key, val in config.items()
+            if key.upper() not in ["VERSION", "MODEL_FILE", "HANDLER_FILE", "EXTRA_FILES", "REQUIREMENTS"]
```

Comment:
Is `SOURCE_VOCAB` missing?

Reply:
These arguments are specific to `torch-model-archiver`. I couldnt find `SOURCE_VOCAB` in https://github.com/pytorch/serve/blob/master/model-archiver/README.md . Per my understanding, the vocabulary file will be pushed as `extra_files` argument.

Reply:
Is this doc old?

https://pytorch.org/serve/model-archiver.html

Reply:
Could you verify that by running `torch-model-archiver --help`?

Reply:
yeah.. looks like it is old .. I checked the help command .. couldnt find source_vocab

```
usage: torch-model-archiver [-h] --model-name MODEL_NAME --serialized-file
                            SERIALIZED_FILE [--model-file MODEL_FILE]
                            --handler HANDLER [--extra-files EXTRA_FILES]
                            [--runtime {python,python2,python3}]
                            [--export-path EXPORT_PATH]
                            [--archive-format {tgz,no-archive,default}] [-f]
                            -v VERSION [-r REQUIREMENTS_FILE]
torch-model-archiver: error: the following arguments are required: --model-name, --serialized-file, --handler, -v/--version
```

Reply:
Thanks!